cmd/temporalite/ui.go

```diff
@@ -33,22 +30,13 @@ func newUIOption(frontendAddr string, uiIP string, uiPort int, codecEndpoint str
 	return temporalite.WithUI(uiserver.NewServer(uiserveroptions.WithConfigProvider(cfg))), nil
 }
 
-func newUIConfig(frontendAddr string, uiIP string, uiPort int, codecEndpoint string, configDir string) (*uiconfig.Config, error) {
-	cfg := &uiconfig.Config{
-		Host: uiIP,
-		Port: uiPort,
-		Codec: uiconfig.Codec{
-			Endpoint: codecEndpoint,
-		},
-	}
+func newUIConfig(cfg *uiconfig.Config, configDir string) (*uiconfig.Config, error) {
 	if configDir != "" {
 		if err := provider.Load(configDir, cfg, "temporalite-ui"); err != nil {
 			if !strings.HasPrefix(err.Error(), "no config files found") {
 				return nil, err
 			}
 		}
 	}
-	cfg.TemporalGRPCAddress = frontendAddr
-	cfg.EnableUI = true
```

Comment:
I am not really sure why those two options where set after having loaded the file based config. The behavior looked inconsistent to me, either the config file overrides the CLI flags values or the opposite, but doing both sounds confusing.

My manual test seem to show I can still load the UI and the UI can still reach out to the frontend 

![image](https://user-images.githubusercontent.com/406923/204768775-fa9cd171-4818-47c5-94e7-3e832d9e751e.png)

when I start `temporalite` with 

```
./temporalite start \
        --namespace default \
        --log-format pretty \
        --ephemeral \
        --ui-codec-endpoint 'https://<the-remote-endpoint>'
```

And the `--headless` flag is still respected.

Maybe I missed something though.

Reply:
> either the config file overrides the CLI flags values or the opposite, but doing both sounds confusing

I would not be surprised if the config file required the frontend address and this just set it back to whatever dummy value was set by a user. Or maybe there is a common/default UI config used in one of our projects that this is explicitly overriding values of. Or maybe our docker compose or helm charts just use some config default template that puts some values here by default we choose to override for Temporalite. At this point you can't be sure that people aren't relying on these two fields to be overridden back to their Temporalite expectation.

Reply:
Ok, right indeed, it's probably safer. Fixing.